prosemirror/model/content.py

```diff
@@ -1,20 +1,61 @@
 import re
+from dataclasses import dataclass, field
 from functools import cmp_to_key, reduce
-from typing import ClassVar
+from typing import (
+    TYPE_CHECKING,
+    ClassVar,
+    Dict,
+    List,
+    Literal,
+    NamedTuple,
+    NoReturn,
+    Optional,
+    TypedDict,
+    Union,
+    cast,
+)
 
 from .fragment import Fragment
 
+if TYPE_CHECKING:
+    from .node import Node
+    from .schema import NodeType
 
+
+@dataclass
+class MatchEdge:
+    type: "NodeType"
+    next: "ContentMatch"
+
+
+@dataclass
+class WrapCacheEntry:
+    target: "NodeType"
+    computed: Optional[List["NodeType"]]
+
+
+class Active(TypedDict):
+    match: "ContentMatch"
+    type: Optional["NodeType"]
+    via: Optional["Active"]
+
+
+@dataclass(eq=False)
```

Comment:
What's the motivation for using dataclasses?

Reply:
I thought mypyc would optimize these better, but turns out that's not the case. I've removed this for now.